test/e2e/launch.e2e.test.js

```diff
@@ -1,108 +1,84 @@
-
-describe('Application launch', function () {
-  this.timeout(60000)
+test.describe.serial('Application launch', async () => {
   let app = null
 
-  afterEach(async function () {
-    if (app && app.isRunning()) {
-      await app.stop()
+  test.afterEach(async () => {
+    if (app) {
+      await app.close()
     }
   })
 
-  async function startApp ({
-    home = tmp.dirSync({ prefix: 'tmp_HOME_', unsafeCleanup: true }).name,
-    repoPath = path.join(home, '.ipfs')
-  }) {
-    app = new Application({
-      path: electronPath,
-      args: ['-r', path.join(__dirname, 'utils/include.js'), path.join(__dirname, '../../src/index.js')],
-      env: {
+  async function startApp ({ repoPath } = {}) {
+    const home = tmp.dirSync({ prefix: 'tmp_home_', unsafeCleanup: true }).name
+    if (!repoPath) {
+      repoPath = path.join(home, '.ipfs')
+    }
+    app = await electron.launch({
+      args: [path.join(__dirname, '../../src/index.js')],
+      env: Object.assign({}, process.env, {
         NODE_ENV: 'test',
         HOME: home,
         IPFS_PATH: repoPath
-      }
+      })
     })
-    await app.start()
     return { app, repoPath, home }
   }
 
-  async function daemonReady (app, timeout = 45000) {
-    // TODO: replace this hack with a signal from the app to know when ipfs is ready.
-    // Right now we can't listen for IPC events in spectron (https://github.com/electron/spectron/issues/91)
-    // As a workaround, we look at console output and match on strings :<
-    const tick = 250
-    const ready = (output) => output && output.match(/(?:daemon is running|Daemon is ready|start daemon FINISHED)/)
-    const hasPeerId = (output) => output && output.trim().match(/PeerID is (\w+)$/)
-    let peerId
-    while (true) {
-      const logs = await app.client.getMainProcessLogs()
-      for (const line of logs) {
-        if (process.env.DEBUG) console.log(line)
-        const idMatch = hasPeerId(line)
-        if (idMatch) peerId = idMatch[1]
-        if (ready(line)) {
-          return { peerId }
+  async function daemonReady (app) {
+    const peerId = await app.evaluate(async ({ ipcMain }) => new Promise((resolve, reject) => {
+      ipcMain.on('ipfsd', (status, peerId) => {
+        switch (status) {
+          // NOTE: this code runs inside the main process of electron, so we cannot use
+          // things we've imported outside of this function. The hard coded values can be
+          // found in src/daemon/consts.js.
+          case 3:
+            reject(new Error('starting daemon failed'))
+            break
+          case 2:
+            resolve(peerId)
+            break
         }
-      }
-      await delay(tick)
-      timeout = timeout - tick
-      if (timeout < 0) throw new Error('timeout while waiting for daemon start in daemonReady(app)')
-    }
+      })
+    }))
+
+    return { peerId }
   }
 
-  it('creates a repository on startup', async function () {
+  test('creates a repository on startup', async () => {
```

Comment:
:information_source:  this and below changes are just refactor from `mocha` and `chai` to modern  `@playwright/test`